File: grading.js
// https://www.hackerrank.com/challenges/grading/problem

function nextFiveMultiple(grade) {
  let i = grade;
  do {
    i++;
  } while (i % 5 !== 0);
  return i;
}

const gradingStudents = grades =>
  grades.map(grade => {
    const nextFiveMultipleReturned = nextFiveMultiple(grade);
    switch (true) {
      case grade < 38:
        return grade;
      case nextFiveMultipleReturned - grade < 3:
        return nextFiveMultipleReturned;
      default:
        return grade;
    }
  });

File: grading/grading.clj
(defn getNextMultiple [denominator integer]
  (let [currInt (inc integer)]
    (if (= 0 (mod currInt denominator))
      currInt
      (getNextMultiple denominator currInt))))

(defn gradingStudents [grades]
  (map
   (fn [grade]
     (let [nextFiveMultiple (getNextMultiple 5 grade)
           shouldRetGrade? (or (< grade 38)
                               (>= (- nextFiveMultiple grade) 3))]
       (if shouldRetGrade? grade nextFiveMultiple)))
   grades))